r/src/array_to_vector.cpp

```diff
@@ -180,7 +183,7 @@ class Converter_Date32 : public Converter_SimpleArray<REALSXP> {
   }
 
   Status Ingest_some_nulls(SEXP data, const std::shared_ptr<arrow::Array>& array,
-                           R_xlen_t start, R_xlen_t n) const {
+                           R_xlen_t start, R_xlen_t n, size_t array_index) const {
```

Comment:
Why do these unrelated converters need an additional arg?

Reply:
Because this only gets called from `Converter::Ingest_some_nulls()` so they all need to have the same interface.

Reply:
Perhaps the dispatch could rather be done by `arrow::VisitTypeInline()` but I'd rather do this in a follow up

Reply:
I think it's better to have a uniform interface. I would like a comment indicating that most implementations will ignore the chunk_index and that (IIUC) only the dictionary conversion path currently uses it.